### PR TITLE
fix #9112, improve android screenshot error message on failure

### LINKED ISSUE #9112: Screenshot command for Android is not working with a reverse tcp connection
## Steps to reproduce
Run the command: "screenshot" on a reverse TCP connection, give you the error:
[-] stdapi_ui_desktop_screenshot: Operation failed: 1

## Expected behavior
To capture the screen or save it somewhere.

## Current behavior
Gets the error: [-] stdapi_ui_desktop_screenshot: Operation failed: 1

## System stuff
Tested on : Android 6.0.1 - Linux 3.10.61-8299335 (armv8l) - Samsung J7 dual sim.

### Metasploit version
metasploit v4.16.11-dev

### I installed Metasploit with:
Kali package via apt

### OS
What OS are you running Metasploit on?
Kali Linux 2017.2

### PR DESCRIPTION
**Pull request by timwr**
## Verification

List the steps needed to make sure this thing works

- [x] Land https://github.com/rapid7/metasploit-payloads/pull/272
- [x] Get an android meterpreter session
- [x] `meterpreter> screenshot`
- [ ] ...
- [x] **Verify** the error message makes sense

### CODE REVIEW

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/ui.rb [hunk 1]
```diff
@@ -162,6 +162,11 @@ def cmd_screenshot( *args )
       print_line( "Screenshot saved to: #{path}" )
 
       Rex::Compat.open_file( path ) if view
+    else
+      print_error("No screenshot data was returned.")
+      if client.platform == 'android'
+        print_error("On Android this command can only capture the applications own framebuffer.")
```
Comment (busterb) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/ui.rb]: application's

Comment (timwr) [reply lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/ui.rb]: oops thanks!

**Review by h00die — COMMENTED** [review]
apparently I never clicked "Submit review"...

**Comment by busterb** [comment]
I landed the bits on framework, will make these updates to move this along. Thanks @h00die / @timwr

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/ui.rb [hunk 2]
```diff
@@ -162,6 +162,11 @@ def cmd_screenshot( *args )
       print_line( "Screenshot saved to: #{path}" )
 
       Rex::Compat.open_file( path ) if view
+    else
+      print_error("No screenshot data was returned.")
```
[the remark this answers is not in the document]
Comment (busterb) [reply lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/ui.rb]: I disabled checks for this a while ago: https://github.com/rapid7/metasploit-framework/blob/master/.rubocop.yml#L100

**Comment by busterb** [comment]
f662049b27ae5ea536cbe59868730c9ed79eddd9
1610db482a39d0479502f7aa76ee648bc444babb

**Comment by busterb** [comment]
## Release Notes

Screenshots are restricted with Android's security model. This adds a user-friendly message when taking a screenshot from an Android application fails.

**Pull request merged**